src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -90,26 +83,27 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         }
 
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsBefore && 
+        if (objResult.getsObjectType() == Allocation__c.sObjectType && !TDTM_ProcessControl.checkRecursionState(TDTM_ProcessControl.flag.ALLOC) &&
             (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate)) {
```

Comment:
I'm reviewing these at the moment, and so far haven't found anything that appears to require these. However, they were certainly there for a reason so I'm still drilling into it.